src/core/meshcop/meshcop_tlvs.cpp

```diff
@@ -202,191 +201,140 @@ const char *StateTlv::StateToString(State aState)
     return aState == kReject ? kStateStrings[2] : kStateStrings[aState];
 }
 
-bool ChannelMaskBaseTlv::IsValid(void) const
+bool ChannelMaskTlv::IsValid(void) const
 {
-    const ChannelMaskEntryBase *cur = GetFirstEntry();
-    const ChannelMaskEntryBase *end = reinterpret_cast<const ChannelMaskEntryBase *>(GetNext());
-    bool                        ret = false;
+    uint32_t channelMask;
 
-    VerifyOrExit(cur != nullptr);
-
-    while (cur < end)
-    {
-        uint8_t channelPage;
-
-        VerifyOrExit((cur + 1) <= end && cur->GetNext() <= end);
-
-        channelPage = cur->GetChannelPage();
-
-#if OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_SUPPORT
-        if (channelPage == OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_CHANNEL_PAGE)
-#else
-        if ((channelPage == OT_RADIO_CHANNEL_PAGE_0) || (channelPage == OT_RADIO_CHANNEL_PAGE_2))
-#endif
-        {
-            VerifyOrExit(static_cast<const ChannelMaskEntry *>(cur)->IsValid());
-        }
+    return (ReadChannelMask(channelMask) == kErrorNone);
```

Comment:
Do we support parsing multiple Channel Mask Entries and skip over ones that have an unknown / unsupported Channel Page? I'm thinking of making sure we support forward compatibility with future Channel Page definitions.

Reply:
Yes. We do support multiple Channel Mask Entries and we skip over the ones not supported by radio (we did the same before this PR  - though in the existing code we may accept an ill-formed TLV value when reading it directly from a `Message` but not when parsing it from `Dataset`).

One thing related to this is for all supported channel pages we assume and require that the `mMaskLength` in the `Entry` to be exactly 4 bytes (i.e. a `uint32_t`  channel mask). This is the behavior in existing code and I kept it the same with new PR. Not sure if we may want to update this to be safer (for forward compatibility). We can update the code to allow longer/shorter mask lengths? @jwhui  what do you think?

- We  support page 0 for 2.4GHz band, and I expect `uint32_t` would be good for it (in future).
- Channel page 2 is used in the code for 915MHz (which is custom `OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT`)  but again with `uint32_t` channel mask.
- We also have `OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_SUPPORT` where it defines its own page number and channel ranges but again the `ChannelMaskTlv` code requires and uses `uint32_t` for mask.

Reply:
Yeah, the length check is what triggered my initial thought. If a channel page is defined, then I assume the length of the channel mask is also defined. But if a channel page is unknown, we should try to skip over the channel mask regardless of its length.

Reply:
> a channel page is defined, then I assume the length of the channel mask is also defined.

Okay. This sounds good to me. We assume 4 bytes for channel page 0 (and same for other proprietray supported ones).

> But if a channel page is unknown, we should try to skip over the channel mask regardless of its length.

Yes. That's the behavior. If radio does not know or support the page, we ignore the `Entry` and jump over it whatever its size (and even allow `mMaskLength` being zero). I think the previous (before this PR) code also did the same.